### ISSUE TITLE
HTML <!-- comments --> placed within text alters whitespace

### ISSUE THREAD

**Issue by MCTaylor17** [beautify-web/js-beautify#1301]
# Description
When HTML comments are beautified, it removes the whitespace surrounding the comment.  This is problematic if the HTML comments occur within text blocks such as paragraphs.  

# Input
When a comment exists inside a text block:
```html
<div><p>Hello <!-- Comment -->World</p></div>
```
# Expected
Ideally, I would like comments to always be ignored wherever they occur, like so:
```html
<div>
	<p>Hello <!-- Comment -->World</p>
</div>
```

# Actual
The problem with the above is the removal of the space following `Hello` like so:
```html
<div>
	<p>Hello
		<!-- Comment -->World</p>
</div>
```

## Settings
Example:
```json
{
	"indent_size": 1,
	"indent_char": "	",
	"wrap_line_length": 0,
	"preserve_newlines": true,
	"max_preserve_newlines": 3
}
```

# Note
I'm aware that I could use:
```json
{
	"content_unformatted": ["p","li","td", "...", "..."]
}
```

But this doesn't strike me as an ideal solution.  I think it would be better if there was an option to ignore comment tags specifically.

**Comment by oleggromov** [comment]
I'm having the same issue here.

**Comment by abhishek-kumar-91** [comment]
I want to work on this issue, Can you assign me?

**Comment by bitwiseman** [comment]
@abhishek-kumar-91  No, I will not assign. But you should feel free to work on this.

**Comment by Kyathirao** [comment]
@bitwiseman can I work on this issue?

**Comment by bitwiseman** [comment]
Yes.

**Comment by serhiyupw** [comment]
For example, instead of using regular space characters, you can use &nbsp; HTML entity to represent the whitespace. This way, when the HTML comments are beautified, the whitespace will be preserved.

<p>
  This is some text with an HTML comment:
  
</p>

**Comment by bitwiseman** [comment]
@serhiyupw 
I don't think that applies here.  HTML entity tags don't get interpreted in comments and the issue isn't what is inside the comment tag but its positioning due to elements outside the tag.
